src/System.Text.Primitives/System/Text/InvariantParser_byte.cs

```diff
@@ -15,7 +15,13 @@ public static partial class InvariantParser
         /// <returns>True if parsing is successful; false otherwise.</returns>
 		public static bool TryParse(byte[] utf8Text, int index, out byte value, out int bytesConsumed)
         {
-            Precondition.Require(utf8Text.Length > 0);
+            // Precondition replacement
+            if (utf8Text.Length < 1 || index < 0 || index > utf8Text.Length)
```

Comment:
`index` needs to be strictly less than utf8Text.Length, since you can only access up to `Length - 1` in an array.